src/include/hip/rpp_hip_common.hpp

```diff
@@ -1825,6 +1825,18 @@ __device__ __forceinline__ void rpp_hip_math_bitwiseOr8(d_float8 *src1_f8, d_flo
         dst_f8->f1[7] = (float)((uchar)(src1_f8->f1[7]) | (uchar)(src2_f8->f1[7]));
 }
 
+__device__ __forceinline__ void rpp_hip_math_exclusiveOr8(d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
+{
+        dst_f8->f1[0] = (float)((uchar)(std::nearbyintf)(src1_f8->f1[0]) ^ (uchar)(std::nearbyintf)(src2_f8->f1[0]));
+        dst_f8->f1[1] = (float)((uchar)(std::nearbyintf)(src1_f8->f1[1]) ^ (uchar)(std::nearbyintf)(src2_f8->f1[1]));
+        dst_f8->f1[2] = (float)((uchar)(std::nearbyintf)(src1_f8->f1[2]) ^ (uchar)(std::nearbyintf)(src2_f8->f1[2]));
+        dst_f8->f1[3] = (float)((uchar)(std::nearbyintf)(src1_f8->f1[3]) ^ (uchar)(std::nearbyintf)(src2_f8->f1[3]));
```

Comment:
In the rpp_hip_math_bitwiseOr8() and rpp_hip_math_bitwiseAnd8() above, we don't use a std::nearbyintf. Why for rpp_hip_math_exclusiveOr8()?

Reply:
It is done so as to fix output differences with FP16 compared to other bit depths. This addresses the same. We should possibly add the same for bitwiseand and bitwiseor also to fix output differences between FP16 and other bit depths